nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -538,24 +546,52 @@ def setup_enc_dec_tokenizers(
             special_tokens=special_tokens,
             use_fast=False,
             r2l=decoder_r2l,
+            legacy=decoder_sentencepiece_legacy,
         )
 
         # validate no token is negative for sentencepiece tokenizers
-        for tok_name, tok_library, tok_model in [
-            ("encoder_tokenizer", encoder_tokenizer_library, encoder_tokenizer),
-            ("decoder_tokenizer", decoder_tokenizer_library, decoder_tokenizer),
+        for tok_name, tok_library, tok_model, legacy in [
```

Comment:
Is it necessary to have the legacy support here in the non-megatron NMT model? It would be great to keep the legacy code in minimal amount of places, so whenever we switch off from using the legacy tokenizer we don't have to worry about a lot of old models breaking.

Reply:
All of this is to train any model regardless of Megatron-based or NeMo-based models using the mT5 tokenizer which requires adding special tokens.